### PR TITLE
Fix another tricky bug in mapFromEntries.

### PR DESCRIPTION
**Pull request by laithsakka**
Summary:
In the function mapFromEntries, if any row entry is null we should  set
 mutableSizes[row] = 0, to make sure the final vector is valid.

A problem is that errors that happens while iterator on those rows for a given
array at a given index, can prevent that check to happen for next rows.

For example: consider the array [(null,1), null], we should set sizes[0] = 0,
but we do not because the check not he null will throw and exit the loop.

This diff fix that issue and add a test case.

fix https://github.com/facebookincubator/velox/issues/6929

Differential Revision: D50000146

### CODE REVIEW

File: velox/functions/prestosql/tests/MapFromEntriesTest.cpp [hunk 1]
```diff
@@ -422,3 +422,27 @@ TEST_F(MapFromEntriesTest, unknownInputs) {
...
+  RowVectorPtr rowVector = makeRowVector(
+      {makeNullableFlatVector<int32_t>({std::nullopt, 2}),
+       makeFlatVector<int32_t>({1, 2})});
+  rowVector->appendNulls(1);
+  rowVector->setNull(2, true);
+
+  auto offsets = makeIndices({0});
+  auto sizes = makeIndices({3});
+  // Array [(null,1), (2,2), null]
+  auto arrayVector = std::make_shared<ArrayVector>(
```
Comment (mbasmanova) [inline velox/functions/prestosql/tests/MapFromEntriesTest.cpp:437]: makeArrayVector(offsets, rowVector)

```
  ArrayVectorPtr makeArrayVector(
      const std::vector<vector_size_t>& offsets,
      const VectorPtr& elementVector,
      const std::vector<vector_size_t>& nulls = {}) {
```

File: velox/functions/prestosql/tests/MapFromEntriesTest.cpp [hunk 2]
```diff
@@ -422,3 +422,27 @@ TEST_F(MapFromEntriesTest, unknownInputs) {
   test("try(map_from_entries(array_constructor(null)))");
   test("try(map_from_entries(null))");
 }
+
+TEST_F(MapFromEntriesTest, nullRowEntriesWithSmallerChildren) {
+  // Row vector is of size 3, childrens are of size 2 since row 2 is null.
+  RowVectorPtr rowVector = makeRowVector(
```
Comment (mbasmanova) [inline velox/functions/prestosql/tests/MapFromEntriesTest.cpp:428]: auto

File: velox/functions/prestosql/tests/MapFromEntriesTest.cpp [hunk 3]
```diff
@@ -422,3 +422,27 @@ TEST_F(MapFromEntriesTest, unknownInputs) {
...
+  // Array [(null,1), (2,2), null]
+  auto arrayVector = std::make_shared<ArrayVector>(
+      pool(),
+      ARRAY(ROW({INTEGER(), INTEGER()})),
+      nullptr,
+      1,
+      offsets,
+      sizes,
+      rowVector);
+  VectorPtr result =
```
Comment (mbasmanova) [inline velox/functions/prestosql/tests/MapFromEntriesTest.cpp:445]: auto

File: velox/functions/prestosql/tests/MapFromEntriesTest.cpp [hunk 4]
```diff
@@ -422,3 +422,27 @@ TEST_F(MapFromEntriesTest, unknownInputs) {
...
+      evaluate("try(map_from_entries(c0))", makeRowVector({arrayVector}));
+  result->validate();
```
Comment (mbasmanova) [inline velox/functions/prestosql/tests/MapFromEntriesTest.cpp:447]: Should we also compare the result with "expected"?

File: velox/functions/prestosql/MapFromEntries.cpp [hunk 5]
```diff
@@ -130,7 +132,11 @@ class MapFromEntriesFunction : public exec::VectorFunction {
         const auto size = inputArray->sizeAt(row);
         const auto offset = inputArray->offsetAt(row);
 
-        for (auto i = 0; i < size; ++i) {
+        // We need this loop separated from the next one, because errors in the
```
Comment (mbasmanova) [inline velox/functions/prestosql/MapFromEntries.cpp:135]: I wonder why don't we need to set mutableSizes[row] = 0 when we encounter a problem with the keys?

Comment (laithsakka) [reply velox/functions/prestosql/MapFromEntries.cpp:135]: that would work also i think

**Review by mbasmanova — APPROVED** [review]
Thanks.

**Pull request closed without merging**